wandb/sdk/wandb_artifacts.py

```diff
@@ -1805,15 +1790,185 @@ def _content_addressed_path(md5: str) -> FilePathStr:
         )
 
 
+class AzureHandler(StorageHandler):
+    def can_handle(self, parsed_url: "ParseResult") -> bool:
+        return parsed_url.scheme == "https" and parsed_url.netloc.endswith(
+            ".blob.core.windows.net"
+        )
+
+    def load_path(
+        self,
+        manifest_entry: "ArtifactManifestEntry",
+        local: bool = False,
+    ) -> Union[URIStr, FilePathStr]:
+        assert manifest_entry.ref is not None
+        if not local:
+            return manifest_entry.ref
+
+        path, hit, cache_open = get_artifacts_cache().check_etag_obj_path(
+            URIStr(manifest_entry.ref),
+            ETag(manifest_entry.digest),
+            manifest_entry.size or 0,
+        )
+        if hit:
+            return path
+
+        account_url, container_name, blob_name, query = self._parse_uri(
```

Comment:
I might make the return type here a namedtuple or a dataclass; but I probably also overuse dataclasses. I think the fact that there is an obvious ordering (url before container, etc.) makes it less likely to make a mistake here and less reason for putting them into an object. (note only, no change requested)

Reply:
Agreed on both. I only use tuples when the order of elements is obvious.